ipaserver/plugins/baseuser.py

```diff
@@ -526,6 +567,82 @@ def convert_attribute_members(self, entry_attrs, *keys, **options):
         except KeyError:
             pass
 
+    def handle_subordinate_ids(self, ldap, dn, entry_attrs):
+        """Handle ipaSubordinateId object class
+        """
+        obj_classes = entry_attrs.get("objectclass")
+        new_subuid = entry_attrs.single_value.get("ipasubuidnumber")
+        new_subgid = entry_attrs.single_value.get("ipasubgidnumber")
+
+        # entry has object class ipaSubordinateId
+        # default to auto-assigment of subuids
+        if obj_classes is not None and "ipasubordinateid" in obj_classes:
+            if new_subuid is None:
+                new_subuid = DNA_MAGIC
+
+        # neither auto-assignment nor explicit assignment
+        if new_subuid is None:
+            # nothing to do
+            return False
+
+        # enforce subuid == subgid for now
+        if new_subgid is not None and new_subgid != new_subuid:
+            raise errors.ValidationError(
+                name="ipasubgidnumber",
+                error=_("subgidnumber must be equal to subuidnumber")
+            )
+
+        self.set_subordinate_ids(ldap, dn, entry_attrs, new_subuid)
+        return True
+
+    def set_subordinate_ids(self, ldap, dn, entry_attrs, subuid):
+        """Set subuid value of an entry
+
+        Takes care of objectclass and sibbling attributes
+        """
+        if "objectclass" in entry_attrs:
+            obj_classes = entry_attrs["objectclass"]
+        else:
+            _entry_attrs = ldap.get_entry(dn, ["objectclass"])
+            entry_attrs["objectclass"] = _entry_attrs["objectclass"]
+            obj_classes = entry_attrs["objectclass"]
+
+        if "ipasubordinateid" not in obj_classes:
+            obj_classes.append("ipasubordinateid")
+
+        if subuid == DNA_MAGIC:
+            subuid = self._fake_dna_plugin(ldap, dn, entry_attrs)
```

Comment:
I assume you'll want a comment around this too like XXX HACK

Reply:
+1